app/src/main/java/org/kiwix/kiwixmobile/zim_manager/fileselect_view/ZimFileSelectFragment.kt

```diff
@@ -111,6 +111,12 @@ class ZimFileSelectFragment : BaseFragment() {
     if (savedInstanceState != null && savedInstanceState.getBoolean(WAS_IN_ACTION_MODE)) {
       zimManageViewModel.fileSelectActions.offer(FileSelectActions.RestartActionMode)
     }
+
+    zimManageViewModel.currentPage.map { it == 1 }.subscribe { if (it) finishActionMode() }
```

Comment:
I think filter would be apt here.

Reply:
I also want this logic in the viewmodel

Reply:
@macgills if I put it in the view model, do I need to make it like the other disposables which are already created there? 
Sorry for a bit of inefficiency from my side here, I'm not very familiar with how to use publish processors, but I'm interested in learning about them as they seem better than live data.

Reply:
They are very different things, livedatas have really only one android specific use and Rx has almost every use under the sun and has implmentations across many languages. It is slightly falling out of popularity in favour of coroutines because it is very complicated and easy to misuse.

So we already have our input Processor, on page change send the page.
Then inside our ViewModel we can have our output processor eg
```
input.map { /*whatever we want to do*/ }
```
we can preform any dat transformation here.

filter only allows an item downstream if the condition inside is true

eg
```
Flowable.just("a", "b", "c")
.filter { it == "b"}
.subscribe { print(it) }
```
would print "b" but without the filter would print "a" then "b" then "c"

Reply:
@macgills please take a look at the latest changes I made. 
I am facing one issue here. When the action mode is on and I shift fragments, the action mode stays on, when I go back to the offline files section and then again to the online section, then the action mode turns off. It doesn't turn off the first time the transition is done. I checked the logs and the function in the view model is called whenever the selected page index is 1, but action mode doesn't get turned off. Could you please provide some help with this?